src/sentry/event_manager.py

```diff
@@ -1139,10 +1139,21 @@ def _save_aggregate(event, hashes, release, metadata, received_timestamp, **kwar
 
     is_new = False
 
-    if root_hierarchical_hash is not None:
-        new_hashes = []
-    else:
+    if root_hierarchical_grouphash is None:
+        # No hierarchical grouping was run, only consider flat hashes
         new_hashes = [h for h in flat_grouphashes if h.group_id is None]
+    elif root_hierarchical_grouphash.group_id is None:
+        # The root hash is not assigned to a group. This can have two reasons:
+        if existing_grouphash.hash in hashes.hierarchical_hashes:
+            # The group was previously split and is now associated with a deeper
+            # hierarchical hash
+            new_hashes = []
```

Comment:
This works for now, but what if we develop a new hierarchical algorithm and we run the original hierarchical algorithm as secondary? Should we check for `found_split` instead?

Reply:
i would hope/expect that flat-hashes still points us towards the right group. do you think this would be testable? you could probably register a new hidden derived strategy configuration for testing in `src/sentry/grouping/strategy/configurations.py`